api/cpp/include/slint.h

```diff
@@ -1091,17 +1091,27 @@ class Repeater
         };
         std::vector<RepeatedInstanceWithState> data;
         private_api::Property<bool> is_dirty { true };
+        std::shared_ptr<Model<ModelData>> model;
 
         void row_added(size_t index, size_t count) override
         {
             is_dirty.set(true);
             data.resize(data.size() + count);
             std::rotate(data.begin() + index, data.end() - count, data.end());
+            for (std::size_t i = index; i < data.size(); ++i) {
+                // all the indexes are dirty
+                data[i].state = State::Dirty;
```

Comment:
This is just yet another bug that got spotted by another test, as the row_changed no longer marked the item as dirty after this patch.


(Note that C++ is not affected by bug #3740 because the C++ listview don't do the "allocate only visible" optimization.  But it's the other side bug  (`model[i]=...`) that needed a fix